Updated Webpacker Initializer Docs (#107)
forem/forem-docs: docs/frontend/javascript-approaches/packs.md

sidebar_position: 2
---

:::important

> Any refactored or new functionality that lives in the asset pipeline, should now be moved to Packs and Webpacker.

Reviewer: Base don my first comment maybe we can update this a bit, I'm not sure that I completely follow what this means.

- [Enhance Jest configuration with Module Aliases](https://alexjover.com/blog/enhance-jest-configuration-with-module-aliases/)
- [import-js/eslint-plugin-import](https://github.com/import-js/eslint-plugin-import/tree/main/resolvers/webpack)

> ## Initializers in Webpacker

Reviewer: I think that it may be useful to write a sentence on what an initializer is and why you'd create one.

> Initializers, that formally lived within the asset pipeline, now reside in Webpack. 

Reviewer: ```suggestion
Initializers, that formerly lived within the asset pipeline, now reside in Webpack. 
```

```
app/javascript/packs/initializers
```

> Upon an initializer being created, it can then be added to the `baseInitializers.js` file to be initialized upon load, similiar to the `initializePage.js` in the asset pipeline.

Reviewer: Feel free to ignore this, I just tweaked it to be a bit more clear

```suggestion
When creating an initializer, you can add it to  `baseInitializers.js` file which will initialize the code upon load. This is  similar to the way `initializePage.js` behaves in the asset pipeline.
```

> 

Reviewer: ```suggestion

Here are [some examples of initializers that currently exist](https://github.com/forem/forem/blob/main/app/javascript/packs/baseInitializers.js#L11-L15). 
```
